include/gul17/expected.h

```diff
@@ -106,14 +106,6 @@ class unexpected;
 #define GUL17_EXPECTED_EXCEPTIONS_ENABLED
 #endif
 
-#if (defined(__GNUC__) && __GNUC__ == 5 && __GNUC_MINOR__ <= 4 && !defined(__clang__))
```

Comment:
This is in line with

```
Drop support for GCC 5.4

[why]
It does not support C++17 anyhow.
```

But how do we make certain that a user is not using GCC 5.4?
I know it is unlikely, but did you add any enforcement (i.e. check compiler version somewhere else and abort if 5.5 or below)?

Reply:
If we want to warn users of old compilers, I would prefer
```cpp
#if __cplusplus < 201703L
```
over specific tests for gcc, clang, msvc, etc.

But where to put it? We do not know which .cc file gets compiled first or which header gets included first. And doing the check in every header seems excessive.

Reply:
I wonder how much is actual compiler supporting some feature and what is relying on standard-library implementation. At least for GCC and Clang this is usually inter-connected, since the compiler build also provides the stl implementation.

Reply:
I have added a sentence to the README. On the website we already have this:
> GUL17 requires at least C++17. It works fine with newer versions of the standard, but might use its own backport types (e.g. [gul17::expected](https://gul-cpp.github.io/gul17/classgul17_1_1expected.html)) in function interfaces.